### DISCUSSION TITLE
Request for Feedback: Eating food while in a Royale hazard?

### DISCUSSION THREAD

**Discussion by bvanvugt** [BattlesnakeOfficial/feedback#63] [Announcements & News]
As reported by **Hegberg** in Discord:

> Just noticed this, is it intentional for when eating food in the hazard, that you health is set to 85 and not 100? @Admin , Seems odd that you don't experience the regular -1 decay, but you do experience the hazard -15 decay on a tile when you eat while in hazard

Note that this is intended behavior, as per the current royale rules order of operations:
1. Reduce health by 1 (for movement)
2. If on food, set health to 100
3. If head in hazard, decrease health by 15

#### However! We'd like to ask what the community thinks about this...

We're inclined to change this behavior, such that eating while in a hazard always sets your Battlesnake to max health (100). We think this is:

* a) more aligned with expected behaviour, and also 
* b) more rewarding for Battlesnakes willing to dip into hazard squares to eat.

Our team thinks these notes combined will make the game more fun to watch, and easier to understand for novice players.

**Note:** No ruleset changes will be made until AFTER July 31, 2021 once Summer League is wrapped up.

**Comment by coreyja** [comment]
> a) more aligned with expected behaviour, and also

Agreed, took me a bit of watching games to realize this was the order of operations

> b) more rewarding for Battlesnakes willing to dip into hazard squares to eat

I think _part_ of the fun of Royale is the danger of the sauce, so I haven't ever been opposed to a saucey food being 'worse' than a normal food. But I do see how it might make more snakes less likely to venture in the sauce 🤷 

That said, not opposed to the change at all! Would be interesting to see how it affects game play!

> Note: No ruleset changes will be made until AFTER July 31, 2021 once Summer League is wrapped up.

Thank you lol!

**Reply by bvanvugt** [reply]
Yeah you make a good point about saucey food should be worse. I think we'd like to control that using the damage caused by sauce, not necessarily the order of operations (if that makes sense).

**Reply by coreyja** [reply]
> I think we'd like to control that using the damage caused by sauce, not necessarily the order of operations (if that makes sense)

This is likely a very small nit pick, but edge cases are fun 😆 

That doesn't exactly make the saucey food worse, it just makes the squares around it worse lol. I can come up with a worst case scenario where a saucey food holds the same value as a normal one, given this rule change

<img width="264" alt="image" src="https://user-images.githubusercontent.com/964771/123318915-bc8b6b00-d4fd-11eb-94f3-5a1840996066.png">

Legend: Pink Sauce. Orange Food. Green snake

If I am in the sauce and have two food options, one out of the sauce and one still in it, this change would make them both have the same 'value' to me. I can get them both in a single move AND get back out of the sauce after eating food.

I'm not saying that this is necessarily bad! Just pointing out a potential edge case lol

**Reply by InterestingBrainPoops** [reply]
This is an edge case, however, as of right now if a snake wished to remain in the hazard sauce for as long as possible after eating a food in the hazard sauce, they can only stay in there for 5 turns. This could pose an issue if there was an advantage to remaining in the sauce (normally there isnt, but still).

**Comment by InterestingBrainPoops** [comment]
This seems like a good idea to me! 
> Our team thinks these notes combined will make the game more fun to watch, and easier to understand for novice players.

Definitely, I think the more intuitive it is, the better. This is kind of like the change from api v0 to v1 with the y axis invert.

**Comment by aubreyyan** [comment]
+1 for making things more consistent and less random
